wurst/systems/spells/gatherer/TeleGathering.wurst

```diff
@@ -0,0 +1,157 @@
+        this.duration = duration
+        this.teleTarget = teleTarget
+        this.sortingPositions = new IterableMap<int, SortingPosition>()
+        startBuffPolling()
+
+    function startBuffPolling()
+        this.pollingLoop = doPeriodically(1) cb ->
+            this.lifetime += 1
+            if this.lifetime >= this.duration
+                or not caster.hasAbility(CASTER_BUFF_ID)
+                or not teleTarget.hasAbility(TARGET_BUFF_ID)
+                destroy this
+
+    function canGatherItem(int itemType) returns bool
+        if itemType == 'XRAW' //Raw meat
+            return false
+
+        //Skip herbs if not herb-gatherer
+        if not this.casterEntity instanceof HerbMaster
+            and not this.casterEntity instanceof Omnigatherer
+                if itemType == ITEM_RIVER_ROOT
+                    or itemType == ITEM_RIVER_STEM
+                    or itemType == ITEM_BUTSU
+                    or itemType == ITEM_BLUE_HERB
+                    or itemType == ITEM_ORANGE_HERB
+                    or itemType == ITEM_YELLOW_HERB
+                    or itemType == ITEM_PURPLE_HERB
+                    or itemType == ITEM_MUSHROOM
+                    return false
+
+        return true
+
+    function handleTeleGather(item itm)
+        let itemType = itm.getTypeId()
+
+        if not canGatherItem(itemType)
+            return
+
+        let sortingPos = this.getSortingPos(itemType)
+        this.caster.removeItem(itm)
+        itm.setPos(sortingPos.pos)
+
+    function getSortingPos(int itemType) returns SortingPosition
+        let targetPos = teleTarget.getPos()
+
+        //If a cached pos exists, return it
+        if this.sortingPositions.has(itemType)
+            return this.sortingPositions.get(itemType)
+
+        //Otherwise find and create a new pos for item type
+        let newPos = new SortingPosition(findPosForItem(itemType))
+        sortingPositions.put(itemType, newPos)
+        return newPos
+
+    function findPosForItem(int itemType) returns vec2
+        let defaultPos = teleTarget.getPos()
+
+        //Look for similar nearby item
+        let nearest = findNearestItem(defaultPos, sortDistanceOffset * MAX_SORTING_SEARCH_LOOPS * 1.1, i -> i.getTypeId() == itemType)
+        if nearest != null
+            return nearest.getPos()
+
+        //Look for a new position around the fire
+        for distanceOffset = this.sortDistanceOffset to MAX_SORTING_SEARCH_LOOPS * SORT_DISTANCE_INCREMENT step SORT_DISTANCE_INCREMENT
+            for angleOffset = this.nextSortAngle to 360 step SORT_ANGLE_INCREMENT
+                this.sortDistanceOffset = angleOffset
+                this.nextSortAngle = angleOffset
+                let maybeTargetPos = defaultPos.polarOffset(angleOffset.toReal().fromDeg(), distanceOffset.toReal())
+
+                if maybeTargetPos.isTerrainWalkable()
+                    return maybeTargetPos
+
+        return defaultPos
```

Comment:
Why not just `return teleTarget.getPos()` here? I don't think there's much benefit of creating a local only used once.